lib/saluki-core/src/components/renderers/context.rs

```diff
@@ -0,0 +1,74 @@
+use memory_accounting::{ComponentRegistry, MemoryLimiter};
+use saluki_health::{Health, HealthRegistry};
+use tokio::runtime::Handle;
+
+use crate::{components::ComponentContext, topology::interconnect::EventStream};
+
+/// Renderer context.
+pub struct RendererContext {
+    component_context: ComponentContext,
+    events: EventStream,
+    memory_limiter: MemoryLimiter,
+    health_handle: Option<Health>,
+    health_registry: HealthRegistry,
+    component_registry: ComponentRegistry,
+    thread_pool: Handle,
+}
+
+impl RendererContext {
+    /// Creates a new `RendererContext`.
+    pub fn new(
+        component_context: ComponentContext, events: EventStream, memory_limiter: MemoryLimiter,
+        component_registry: ComponentRegistry, health_handle: Health, health_registry: HealthRegistry,
+        thread_pool: Handle,
+    ) -> Self {
+        Self {
+            component_context,
+            events,
+            memory_limiter,
+            health_handle: Some(health_handle),
```

Comment:
Do we need to warp it into an Option? It seems that we then unwrap it or panic on line 42.

Reply:
We do want it to be `Option<T>` in this case, because it should only be taken once... and we're just consolidating that logic/behavior by doing it all (including the `expect` call) in `take_health_handle` instead of making every component do it themselves. [^1]

[^1]: All component context types have a `take_health_handle` method.